### PR TITLE
Drag and Drop for Global Awareness

### PR DESCRIPTION
**Pull request by sbatten**
Drafting out what it should look like and testing its application

I think the observer should support registering participants (elements that can be dragged and register a set of callbacks) and listeners (just a set of callbacks for listening to global drag start and end).

Participants may need to listen to both their drag start event and others', should they be separate callbacks or should the participant handle checking?

We can also probably group the view id and type logic into the observer so it is responsible for populating and unpopulating the LocalTransferData

### CODE REVIEW

File: src/vs/workbench/browser/parts/activitybar/media/activityaction.css [hunk 1]
```diff
@@ -9,6 +9,32 @@
...
+	content: '';
+	width: 48px;
+	height: 2px;
+	display: block;
+	background-color: var(--insert-border-color);
+	margin-top: 1px;
+	margin-bottom: -3px;
+}
+
+/* .monaco-workbench .activitybar > .content :not(.monaco-menu) > .monaco-action-bar .action-item.insertAfter {
```
Comment (sbatten) [inline src/vs/workbench/browser/parts/activitybar/media/activityaction.css]: remove commented code

File: src/vs/workbench/browser/parts/activitybar/media/activitybarpart.css [hunk 2]
```diff
@@ -7,6 +7,18 @@
...
+.monaco-workbench .part > .drag-overlay {
+	transition-property: opacity;
+	transition-duration: .2s;
+	width: 100%;
+	height: 100%;
+	position: absolute;
+	opacity: 0;
+	top: 0;
+	pointer-events: none;
+	/* z-index: 1; */
```
Comment (sbatten) [inline src/vs/workbench/browser/parts/activitybar/media/activitybarpart.css]: remove commented code

File: src/vs/workbench/browser/parts/sidebar/sidebarPart.ts [hunk 3]
```diff
@@ -154,6 +152,30 @@ export class SidebarPart extends CompositePart<Viewlet> implements IViewletServi
 
 		super.create(parent);
 
+		const overlay = document.createElement('div');
+		addClass(overlay, 'drag-overlay');
+		parent.appendChild(overlay);
+
+		CompositeDragAndDropObserver.INSTANCE.registerTarget(this.element, {
+			onDragStart: e => {
+				// this.element.style.outline = `1px solid`;
```
Comment (sbatten) [inline src/vs/workbench/browser/parts/sidebar/sidebarPart.ts]: remove commented code

**Review by sandy081 — APPROVED** [review]
Did a brief look and overall the code structure makes sense to me.

**Pull request merged**